### ISSUE TITLE
Wasteful `model.train()` at eval time during `compute_eval_losses=True` training

### ISSUE THREAD

**Issue by Andrewwango** [deepinv/deepinv#1096]
If I want to do self-supervised early-stopping, I need to set `compute_eval_losses=True` in `deepinv.Trainer` to have the loss values for early stopping. In `Trainer.compute_loss`, I don't know why we set `train=True` at eval time:

```python
        if train or self.compute_eval_losses:
            # Evaluate reconstruction network
            x_net = self.model_inference(y=y, physics=physics, x=x, train=True)
```

It makes sense during training, but during eval, this calls model inference with grads, which is wasteful if we only want the loss values (for computing at eval time). But we do need `model` to be in training mode (for loss-adapted models that have different performance at train vs eval time). So we need to tell `model_inference` somehow to put the model in training mode but to do `no_grad`. Maybe we wrap `no_grad` in `compute_loss` if `compute_eval_losses=True`?
